tests/e2e/utils/DriverHelper.ts

```diff
@@ -426,24 +492,33 @@ export class DriverHelper {
     }
 
     public async waitAndGetText(elementLocator: By, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT): Promise<string> {
-        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
         const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
 
         Logger.trace(`DriverHelper.waitAndGetText ${elementLocator}`);
 
         for (let i = 0; i < attempts; i++) {
-            const element: WebElement = await this.waitVisibility(elementLocator, timeout);
-
             try {
```

Comment:
2 levels of try/catch

Reply:
I had issues for the methods failing after times that did not match the timeout without a sensual exception message.
The waitVisibility method was also failing with ElementNotFound that was never caught and caused issue with the logic of the tests.
The second layer of try-catch is there to prevent the method failing without reaching the desired amount of attempts/timeout and if the issue persists, it throws the correct exception.